code/API/server.js

```diff
@@ -95,34 +100,62 @@ app.get('/journals/:id', async (req, res) => {
 })
 
 
-app.post('/createJournal', async (req, res) => {
-    await pool.connect()
+app.post('/createJournal', authenticateToken, async (req, res) => {
+    await pool.connect();
 
-    const entryText = req.body.entryText
-    const keywords = req.body.keywords // to be in the format "keyword1,keyword2,keyword3" no spaces between commas
-    const mood = req.body.mood
-    const userId = req.body.userId
+    const { entryText, keywords, mood, userId, year, month, day } = req.body;
 
     try {
-        const text = 'INSERT INTO journalentries (user_id, entry_text, keywords, mood) VALUES ($1, $2, $3, $4)'
-        const values = [userId, entryText, keywords, mood]
-
-        await pool.query(text, values)
-        res.status(200).send({body: 'Journal Created'})
+        console.log('Year:', year, 'Month:', month, 'Day:', day);
+        const entryDate = new Date(year, month - 1, day); // Adjust month since JavaScript Date months start at 0
```

Comment:
Shouldn't this be `month + 1` if the JS months start at 0? Otherwise you will get a -1

Reply:
It is -1 because JS is 0 indexed, while the calendar in the app is 1 indexed meaning it starts at 1. So if February is selected in the app, the app would send a 2, and JS would understand it as March if it is not -1